spec/mailers/previews/candidate_mailer_preview.rb

```diff
@@ -266,6 +258,18 @@ def conditions_not_met
 
 private
 
+  def candidate
+    candidate = FactoryBot.build_stubbed(:candidate)
+
+    # This is not great. It's necessary because some of our mail templates
+    # generates and send a new magic link token to candidates.
+    def candidate.update!(*)
```

Comment:
How does this work? Are you monkey patching the instance of candidate to override `.update()`? I assumed thats what this is doing and I understand 'why' but have never seen this syntax before so I'm not sure how 🤔?

Reply:
Yup, but it's not a real candidate object, it's a `build_stubbed`. `update!` is already monkey patched to something that raises. For the purpose of the mailer previews, I want it to carry on. 🤔